cudamapper/src/main.cu

```diff
@@ -137,46 +146,43 @@ int main(int argc, char* argv[])
         std::vector<std::pair<std::int32_t, int32_t>> target_ranges;
     };
 
+    ///Factor of 1000000 to make max cache size in MiB
+    auto query_chunks  = query_parser->get_read_chunks(index_size * 1000000);
+    auto target_chunks = target_parser->get_read_chunks(target_index_size * 1000000);
+
     //First generate all the ranges independently, then loop over them.
     std::vector<query_target_range> query_target_ranges;
 
-    for (std::int32_t query_start_index = 0; query_start_index < queries; query_start_index += index_size)
+    int target_idx = 0;
+    for (auto const& query_chunk : query_chunks)
     {
-
-        std::int32_t query_end_index = std::min(query_start_index + index_size, queries);
-
-        query_target_range q;
-        q.query_range = std::make_pair(query_start_index, query_end_index);
-
-        std::int32_t target_start_index = 0;
-        // If all_to_all mode, then we can optimzie by starting the target sequences from the same index as
-        // query because all indices before the current query index are guaranteed to have been processed in
-        // a2a mapping.
-        if (all_to_all)
+        query_target_range range;
```

Comment:
I cannot comment on lines which were not edited. Can you rename `query_target_range` to `QueryTargetRange` as it is a struct?